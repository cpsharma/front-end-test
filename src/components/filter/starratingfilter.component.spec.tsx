import { h } from "preact";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-preact-pure";
import StarRatingFilterComponent, {
  StarRatingFilterProps,
} from "./starratingfilter.component";

configure({ adapter: new Adapter() });

describe("StarRatingFilter Component", () => {
  const props: StarRatingFilterProps = {
    starRatings: ["3", "4", "5"],
    selectedStarRatings: [],
    handleStarRatingFilter: (event: Event): void => {},
    label: "Start Rating Filter",
    id: "star-rating-filter",
  };
  const star_rating_component = mount(<StarRatingFilterComponent {...props} />);

  it("renders properly", () => {
    expect(star_rating_component).toMatchSnapshot();
  });

  it("should render star filters", async () => {
    expect(star_rating_component.find("li").length).toBe(3);
  });

  it("should render correct stars", async () => {
    expect(star_rating_component.find("li").at(0).find(".fa-star").length).toBe(
      Number(props.starRatings[0])
    );
    expect(star_rating_component.find("li").at(1).find(".fa-star").length).toBe(
      Number(props.starRatings[1])
    );
    expect(star_rating_component.find("li").at(2).find(".fa-star").length).toBe(
      Number(props.starRatings[2])
    );
  });

  it("should select star filters", async () => {
    const props = {
      starRatings: ["3", "4", "5"],
      selectedStarRatings: ["3"],
      handleStarRatingFilter: (event: Event): void => {},
      label: "Start Rating Filter",
      id: "star-rating-filter",
    };
    const star_rating_component = mount(
      <StarRatingFilterComponent {...props} />
    );
    expect(star_rating_component.find("li").at(0).find(".fa-star").length).toBe(
      Number(props.starRatings[0])
    );
  });
});
