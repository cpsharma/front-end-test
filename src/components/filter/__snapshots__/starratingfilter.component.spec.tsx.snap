// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRatingFilter Component renders properly 1`] = `ReactWrapper {}`;
